app/controllers/DeclarationController.scala

```diff
@@ -89,15 +93,20 @@ class DeclarationController @Inject()(
     }
   }
 
-  private def createCase(newCaseRequest: NewCaseRequest, attachments: Seq[PublishedFileAttachment],
-                         letter: Option[PublishedFileAttachment], answers: UserAnswers)
-                        (implicit headerCarrier: HeaderCarrier): Future[Case] = {
-    caseService.create(appendAttachments(newCaseRequest, attachments, letter, answers))
+  private def createCase(
+    newCaseRequest: NewCaseRequest,
+    attachments: Seq[PublishedFileAttachment],
+    confidentialityStatuses: Map[String, Boolean],
+    letter: Option[PublishedFileAttachment], answers: UserAnswers
+  )(implicit headerCarrier: HeaderCarrier): Future[Case] = {
+    caseService.create(appendAttachments(newCaseRequest, attachments, confidentialityStatuses, letter, answers))
   }
 
-  private def appendAttachments(caseRequest: NewCaseRequest, attachments: Seq[PublishedFileAttachment],
+  private def appendAttachments(caseRequest: NewCaseRequest,
+                                attachments: Seq[PublishedFileAttachment],
+                                confidentialityStatuses: Map[String, Boolean],
                                 letter: Option[PublishedFileAttachment], answers: UserAnswers): NewCaseRequest = {
-    def toAttachment: PublishedFileAttachment => Attachment = file => Attachment(file.id)
+    def toAttachment: PublishedFileAttachment => Attachment = file => Attachment(file.id, confidentialityStatuses(file.id))
```

Comment:
This map lookup `confidentialityStatuses(file.id)` could throw an exception if the key doesn't exist, should we handle this case?

Reply:
This is an interesting one. The structure of the application service flow should always ensure a value is present for the file id, but I guess it's possible it could change in a way which does not mean it's guaranteed.

Reply:
Yeah I think this is similar to the use of `last` and `get` in [MakeFileConfidentialController](https://github.com/hmrc/binding-tariff-trader-frontend/pull/225/files#diff-5ae651b06f87aae2d9ee772e62f1ea46). Both of those calls could throw but only if the user has somehow broken the navigation and followed the journey out of order, and I think that is probably worthy of a 500 in any case.

Reply:
It's similar, but in this instance it'll throw during submission and we'll potentially have completed some steps (e.g. published the attachments but then fail to create the case).

Would it make sense to validate the two maps first in `onSubmit` and we could potentially combine the validated attachment with the confidentiality boolean to avoid subsequent lookups?

e.g. in `onSubmit` ...

```
    val attachments: Seq[FileAttachment] = answers
      .get(SupportingMaterialFileListPage).map(_.fileAttachments)
      .get(UploadSupportingMaterialMultiplePage)
      .getOrElse(Seq.empty)

    val confidentialityStatuses = answers
      .get(MakeFileConfidentialPage)
      .getOrElse(Map.empty[String, Boolean])

    // validate the attachments and confidentialityStatuses, and combine them somehow?
```

Reply:
Yeah I don't see why not, will look at it this afternoon 👍 

Ideally I don't think we would have this fragmented UserAnswers structure and would rather have an application structure that was correct by construction that we could lens into to update but I think that would be a bigger change than I dare to make 😆

Reply:
FWIW the reason I am a bit cagey about updating the `DeclarationController` is that I know it's going to be deleted in DIT-2339 by @colesadam and @warrenskiwork1979's PR for the check your answers page so I have been trying to change it the smallest amount possible so that I can adapt the changes onto their PR later 😱

Reply:
😱  On the plus side it looks like the code for the onSubmit is moved rather than significantly changed